### PR TITLE
CLDR-13948 deploy to cldr-smoke2 on merge

### PR DESCRIPTION
**Pull request by srl295**
CLDR-13948

- restructure the 'ant' workflow, for parallel building
- on push to master, deploy to cldr-ref.

(This is on the main fork because of secrets, so i could try out deployment. )

### CODE REVIEW

**Comment by srl295** [comment]
When this is merged, it will update https://cldr-ref.unicode.org/cldr-apps/

File: .github/workflows/ant.yml [hunk 1]
```diff
@@ -60,10 +57,105 @@ jobs:
...
+      - test
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download cldr-apps.war
+      uses: actions/download-artifact@v2
+      with:
+        name: cldr-apps
+    - name: Deploy to cldr-ref (experimental)
+      # Only on push.
+      if: ${{ github.event_name == 'push' }}
```
Comment (echeran) [inline .github/workflows/ant.yml:133]: Note: this workflow is configured to execute on push events only on the master branch (which is what gets triggered when merging a PR).  Which makes sense, as it does for certain other things (ex: we have our API docs re-generate only on merges to master).

The comment "Only on push" kinda of obfuscates that fact, so maybe reword?

Comment (srl295) [reply .github/workflows/ant.yml:133]: @echeran the point is that PRs don't cause a deployment, only a merge. The workflow is triggered on pr OR push.

Comment (srl295) [reply .github/workflows/ant.yml:134]: @echeran updated the wording of the comment.

**Comment by srl295** [comment]
⚠️ the Branch Protection Rules will need to change when this merges, because it reorganizes some of the checks.  For example, `cldr-ant/test` is a new check, which is not (as I write this) required.

I don't think  removed any checks.

**Comment by srl295** [comment]
@macchiati now approved by @echeran , can i get a rubber stamp?

**Comment by macchiati** [comment]
Glad to, but I think you need to add me as a reviewer.
